Crash report "java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.ClipData$Item android.content.ClipData.getItemAt(int)' on a null object reference"

UUID: ab2083045c8247445ebddb4e7ef02afb
Build flavor: dev
Version name: v4.11.1-f8888cd
OS info: Android 7.0, sdk version: 24
Report description: No title
Logs:
```
E/Kuroba-dev | UNCAUGHT: Called from unhandled exception handler.
E/Kuroba-dev | UNCAUGHT: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=2, result=-1, data=Intent { dat=content://com.android.providers.media.documents/document/image:141436 flg=0x1 }} to activity {com.github.adamantcheese.chan.dev/com.github.adamantcheese.chan.StartActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.ClipData$Item android.content.ClipData.getItemAt(int)' on a null object reference
E/Kuroba-dev | UNCAUGHT: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4196)
E/Kuroba-dev | UNCAUGHT: 	at android.app.ActivityThread.handleSendResult(ActivityThread.java:4239)
E/Kuroba-dev | UNCAUGHT: 	at android.app.ActivityThread.-wrap20(ActivityThread.java)
E/Kuroba-dev | UNCAUGHT: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1599)
E/Kuroba-dev | UNCAUGHT: 	at android.os.Handler.dispatchMessage(Handler.java:102)
E/Kuroba-dev | UNCAUGHT: 	at android.os.Looper.loop(Looper.java:165)
E/Kuroba-dev | UNCAUGHT: 	at android.app.ActivityThread.main(ActivityThread.java:6375)
E/Kuroba-dev | UNCAUGHT: 	at java.lang.reflect.Method.invoke(Native Method)
E/Kuroba-dev | UNCAUGHT: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:912)
E/Kuroba-dev | UNCAUGHT: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:802)
E/Kuroba-dev | UNCAUGHT: Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.content.ClipData$Item android.content.ClipData.getItemAt(int)' on a null object reference
E/Kuroba-dev | UNCAUGHT: 	at com.github.adamantcheese.chan.ui.helper.ImagePickDelegate.onActivityResult(ImagePickDelegate.java:206)
E/Kuroba-dev | UNCAUGHT: 	at com.github.adamantcheese.chan.StartActivity.onActivityResult(StartActivity.java:548)
E/Kuroba-dev | UNCAUGHT: 	at android.app.Activity.dispatchActivityResult(Activity.java:7109)
E/Kuroba-dev | UNCAUGHT: 	at android.app.ActivityThread.deliverResults(ActivityThread.java:4192)
E/Kuroba-dev | UNCAUGHT: 	... 9 more
E/Kuroba-dev | UNCAUGHT: ------------------------------
E/Kuroba-dev | UNCAUGHT: END OF CURRENT RUNTIME MESSAGES
E/Kuroba-dev | UNCAUGHT: ------------------------------
E/Kuroba-dev | UNCAUGHT: Android API Level: 24
E/Kuroba-dev | UNCAUGHT: App Version: v4.11.1-f8888cd
E/Kuroba-dev | UNCAUGHT: Development Build: Yes
E/Kuroba-dev | UNCAUGHT: Phone Model: Xiaomi Redmi Note 4


=== SETTINGS ===
Prefetching enabled: false
Thread downloading enabled: true, active downloads = 0
Hi-res thumbnails enabled: false
Youtube titles parsing enabled: true
Youtube durations parsing enabled: false
Concurrent file loading chunks count: 2
WEBM streaming enabled: false
Saved files base dir info: Java API location: /storage/emulated/0/Kuroba-dev/files, SAF location: , active: Java API
Local threads base dir info: Java API location: /storage/emulated/0/Kuroba-dev/saved_threads, SAF location: , active: Java API
Phone layout mode: PHONE
```

--- a/Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/helper/ImagePickDelegate.java
+++ b/Kuroba/app/src/main/java/com/github/adamantcheese/chan/ui/helper/ImagePickDelegate.java
@@ -17,6 +17,7 @@
 package com.github.adamantcheese.chan.ui.helper;
 
 import android.app.Activity;
+import android.content.ClipData;
 import android.content.Intent;
 import android.content.pm.PackageManager;
 import android.content.pm.ResolveInfo;
@@ -201,10 +202,7 @@ public class ImagePickDelegate {
         boolean canceled = false;
 
         if (resultCode == Activity.RESULT_OK && data != null) {
-            uri = data.getData();
-            if (uri != null) {
-                uri = data.getClipData().getItemAt(0).getUri();
-            }
+            uri = getUriOrNull(data);
             if (uri != null) {
                 Cursor returnCursor = activity.getContentResolver().query(uri, null, null, null, null);
                 if (returnCursor != null) {
@@ -236,6 +234,24 @@ public class ImagePickDelegate {
         }
 
         return true;
+    }
+
+    @Nullable
+    private Uri getUriOrNull(Intent intent) {
+        if (intent.getData() != null) {
+            return intent.getData();
+        }
+
+        ClipData clipData = intent.getClipData();
+        if (clipData == null) {
+            return null;
+        }
+
+        if (clipData.getItemCount() == 0) {
+            return null;
+        }
+
+        return clipData.getItemAt(0).getUri();
     }
 
     private void run() {

